nncf/experimental/onnx/graph/onnx_graph.py

```diff
@@ -191,17 +195,31 @@ def get_nodes_by_type(self, node_type: str) -> List[onnx.NodeProto]:
                 output.append(node)
         return output
 
-    def get_weight_tensor_name(self, node_name: str) -> str:
-        # TODO(kshpv): add search of input weight tensor
+    def get_weight_tensor(self, node: onnx.NodeProto) -> Tuple[str, np.ndarray]:
         """
-        Returns weight tensor name from the 1-index.
+        Returns node's weight tensor name and its value.
 
-        :param node_name: Name of the node.
-        :return: Weight tensor name.
+        :param node: Node, in which the weight tensor finally applied.
+        :return: Weight tensor name and its value.
         """
+        node_name = node.name
         node_inputs = self.get_node_edge_names(node_name)['input']
-        weight_tensor = node_inputs[1]
-        return weight_tensor
+        weight_input = node_inputs[1]
+        try:
+            return self.get_initializer(weight_input).name, self.get_initializers_value(weight_input)
+        except RuntimeError as e:
```

Comment:
Why do we add the `try-except` block here? It looks like a dirty hack for the case. It's better to have code with `if`'s instead of `try-except`'s, IMHO.

Reply:
This is also a relevant to ONNXGraph methods. The get_initializer() raise an Exception whether there is no such Initalizaer in the model. Before when we covered mostly simple cases, it worked fine, but now it doesn't look good.
So, I suggest to create a ticket on refactoring ONNXGraph when we can discuss it

Reply:
From my point of view, we should not add such kind of code that we will definitely refactor in the future.

Reply:
In that case, it demands refactoring of `get_initializer()` and `get_initializers_value()`. Do you suggest to do it in this PR?

Reply:
I see. My concern is that we will become ubiquitously dependent on these functions in more algorithms and never change this behavior at all.

Reply:
No, It will be definitely reviewed during refactoring of ONNXGraph. As there are many concerns about this I will do this when finished more prioritized tasks.

Reply:
ticket 96982